ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/CliCommandFactory.java

```diff
@@ -38,25 +41,28 @@ public static CliCommand getCommandExecutor(final String commandName) throws Com
     }
 
     public static String listAllCommands() {
-        final StringBuilder commands = new StringBuilder("Installed Commands: ");
-        final Iterator<CliCommand> implementations = getAllCommands();
-        while (implementations.hasNext()) {
-            final CliCommand implementation = implementations.next();
-            commands.append(CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, implementation.getClass().getSimpleName()));
-            if (implementations.hasNext()) {
-                commands.append(", ");
-            }
-        }
-        return commands.toString();
+        final StringBuilder commandHelp = new StringBuilder("Installed Commands: ");
+
+        FluentIterable<String> commands = FluentIterable.from(getAllCommands()).transform(
+                new Function<CliCommand, String>() {
+                    @Nullable
+                    @Override
+                    public String apply(@Nullable CliCommand input) {
+                        return CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, input.getClass().getSimpleName());
+                    }
+                }
+        );
+        final Joiner joiner = Joiner.on(", ");
+        commandHelp.append(joiner.join(commands));
+        return commandHelp.toString();
     }
 
-    private static Iterator<CliCommand> getAllCommands() {
+    private static Iterable<CliCommand> getAllCommands() {
         final ServiceLoader<CliCommand> loader =
```

Comment:
Can return instead of initialize
`return ServiceLoader.load(CliCommand.class);`